src/lib.rs

```diff
@@ -1,3 +1,117 @@
+//!    character instead of 1 for ASCII. This would be quite wasteful, so the most popular way to
+//!    store unicode strings (utf-8) uses a continuation-bit style scheme to compress unicode
+//!    characters where possible. Another older scheme is utf-16, which works similarly, but uses 2
+//!    bytes (16 bits) as its smallest size.
+//!  - *grapheme, grapheme cluster* - Outside of computer typesetting, graphemes can have multiple
+//!    meanings in linguistics (see wikipedia). Within computer typesetting (and unicode in
+//!    particular), a grapheme cluster (called grapheme in the rest of these docs) can be thought
+//!    of as the conceptual version of a glyph. One exception to this is Han unification
+//!    (see below). It's important to note that a grapheme cluster may be 1 or more unicode
+//!    characters.  An example of a multi-character grapheme cluster is a rainbow flag (`U+1F3F3
+//!    (white flag), U+200D (zwj), U+1F308 (rainbow)`) which produces a single rainbow flag glyph.
+//!  - *script* - A set of graphemes that are used together to achieve written communcation.
+//!  - *language* - A context for deriving meaning from a script. Collections of graphemes from a
+//!    script will have different meanings in different languages, for example the word "pain"
+//!    means different things in English and French ("bread" in french).
+//!  - *locale* - Contextual information that defines how certain properties of the user interface
+//!    should look. Examples are whether text should be rendered left-to-right, or right-to-left
+//!    (as is the case in Hebrew, for example), and which varaint of Han characters to use.
+//!    Non-text locale information includes date format and timezone.
+//!  - *Han unification, CJK* - A number of Asian languages derive from ancient Chinese. The
+//!    acronym CJK expands to Chinese, Japanese, Korean. Traditional Taiwanese also has this
+//!    property, but since Taiwan has now adopted the Latin script it is less of an issue in
+//!    practice. These languages use different glyphs for the same unicode grapheme (the graphemes
+//!    can be thought of as representing ancient Han Chinese).
+//!  - *RTL, LTR, BiDi* - Some scripts are read from the right to the left (RTL), the opposite of latin
+//!    which is read left-to-right (LTR). If a span of text contains graphemes from both LTR and
+//!    RTL languages, then multiple directions of text will be used, and this is known as
+//!    bi-directional (BiDi) text. For example, if an english sentence contains a hebrew word, a
+//!    reader would expect that the order of letters in that word to be flipped.
+//!  - *cursive* - todo (text rendering hates you says the word cursive describes the situation
+//!    where the position of a grapheme in a word affects which glyph is used).
+//!  - *font shaping* - The act of selecting particular glyphs to represent graphemes based on
```

Comment:
https://harfbuzz.github.io/what-is-harfbuzz.html#what-is-text-shaping is a good reference for either summarizing or pointing to or both.